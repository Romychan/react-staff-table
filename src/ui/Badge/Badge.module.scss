.badge {
  @apply inline-flex items-center rounded-md px-2 py-1 text-xs font-medium ring-1 ring-inset;

  &.default {
    @apply bg-gray-50 text-gray-600 ring-gray-500/10 dark:bg-gray-700 dark:text-gray-300;
  }

  &.green {
    @apply bg-green-50 text-green-700 ring-green-600/20 dark:bg-green-900 dark:text-green-300;
  }

  &.red {
    @apply bg-red-50 text-red-700 ring-red-600/10 dark:bg-red-900 dark:text-red-300;
  }

  &.yellow {
    @apply bg-yellow-50 text-yellow-800 ring-yellow-600/20 dark:bg-yellow-900 dark:text-yellow-300;
  }
}
