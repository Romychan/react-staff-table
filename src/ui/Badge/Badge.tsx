import { PropsWithChildren } from 'react';
import clsx from 'clsx';

import styles from './Badge.module.scss';

interface IBadgeProps {
  colorScheme?: 'default' | 'green' | 'red' | 'yellow';
}

export const Badge = ({
  colorScheme = 'default',
  children,
}: PropsWithChildren<IBadgeProps>) => {
  return (
    <span className={clsx(styles.badge, styles[colorScheme])}>{children}</span>
  );
};
