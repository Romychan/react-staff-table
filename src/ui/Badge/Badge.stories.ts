import type { Meta, StoryObj } from '@storybook/react';

import { Badge } from './Badge';

/** This component can be used to display the status. */
const meta = {
  title: 'UI/Badge',
  component: Badge,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {
    colorScheme: 'default',
  },
  argTypes: {
    colorScheme: {
      control: { type: 'select' },
    },
  },
} satisfies Meta<typeof Badge>;

export default meta;
type Story = StoryObj<typeof meta>;

/** This state is used to display the default status. */
export const Default: Story = {
  args: {
    children: 'Default',
  },
};

/** This state is used to display the success status. */
export const Success: Story = {
  args: {
    children: 'Success',
    colorScheme: 'green',
  },
};

/** This state is used to display the error status. */
export const Error: Story = {
  args: {
    children: 'Error',
    colorScheme: 'red',
  },
};

/** This state is used to display the warning status. */
export const Warning: Story = {
  args: {
    children: 'Warning',
    colorScheme: 'yellow',
  },
};
