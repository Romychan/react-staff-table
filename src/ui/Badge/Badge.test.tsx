import { render, screen } from '@testing-library/react';

import { Badge } from './Badge';

describe('Badge', () => {
  it('should render correctly', () => {
    render(<Badge>Default</Badge>);
    const badgeElement = screen.getByText(/Default/i);

    expect(badgeElement).toBeInTheDocument();
    expect(badgeElement.getAttribute('class')).toMatch(/default/gi);
  });

  it('should render correctly with green color scheme', () => {
    render(<Badge colorScheme="green">Success</Badge>);
    const badgeElement = screen.getByText(/Success/i);

    expect(badgeElement).toBeInTheDocument();
    expect(badgeElement.getAttribute('class')).toMatch(/green/gi);
  });

  it('should render correctly with red color scheme', () => {
    render(<Badge colorScheme="red">Error</Badge>);
    const badgeElement = screen.getByText(/Error/i);

    expect(badgeElement).toBeInTheDocument();
    expect(badgeElement.getAttribute('class')).toMatch(/red/gi);
  });

  it('should render correctly with yellow color scheme', () => {
    render(<Badge colorScheme="yellow">Warning</Badge>);
    const badgeElement = screen.getByText(/Warning/i);

    expect(badgeElement).toBeInTheDocument();
    expect(badgeElement.getAttribute('class')).toMatch(/yellow/gi);
  });

  it('should match snapshot', () => {
    const container = render(<Badge>Default</Badge>);
    expect(container).toMatchSnapshot();
  });
});
