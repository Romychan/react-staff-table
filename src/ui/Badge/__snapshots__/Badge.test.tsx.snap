// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="_badge_63aabb _default_63aabb"
      >
        Default
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="_badge_63aabb _default_63aabb"
    >
      Default
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
